mpa/cls/stage.py

```diff
@@ -204,17 +204,20 @@ def configure_task(cfg, training, model_meta=None, **kwargs):
 
                 if not cfg.model.get('multilabel', False) and not cfg.model.get('hierarchical', False):
                     efficient_mode = cfg['task_adapt'].get('efficient_mode', True)
-                    gamma = 2 if efficient_mode else 3
                     sampler_type = 'balanced'
 
                     if len(set(model_classes) & set(dst_classes)) == 0 or set(model_classes) == set(dst_classes):
                         cfg.model.head.loss = dict(type='CrossEntropyLoss', loss_weight=1.0)
                     else:
+                        cls_num_list = [train_data_cfg.num_data[data_cls] for data_cls in dst_classes]
+                        if cfg.model.head.type == 'NonLinearClsHead':
```

Comment:
This logic is extremely unclear, so I agree with Harim, if introducing a new head can be avoided, it's better to do that. If we can't avoid adding new heads, one can create a config where `NonLinearClsHead` is implicitly written.